cunumeric/eager.py

```diff
@@ -1491,6 +1495,14 @@ def unary_reduction(
             np.argmin(
                 rhs.array, out=self.array, axis=orig_axis, keepdims=keepdims
             )
+        elif op == UnaryRedCode.NANARGMAX:
+            np.nanargmax(
+                rhs.array, out=self.array, axis=orig_axis, keepdims=keepdims
+            )
+        elif op == UnaryRedCode.NANARGMIN:
+            np.nanargmin(
+                rhs.array, out=self.array, axis=orig_axis, keepdims=keepdims
+            )
```

Comment:
I feel all four cases for arg reduction can be tucked into the general one at line 1475. the only extra bit that needs to be done is to make sure `where` doesn't get passed when it's one of the arg reductions.